lib/Doctrine/Common/Proxy/Autoloader.php

```diff
@@ -49,7 +49,11 @@ public static function resolveFile($proxyDir, $proxyNamespace, $className)
             throw InvalidArgumentException::notProxyClass($className, $proxyNamespace);
         }
 
-        $className = str_replace('\\', '', substr($className, strlen($proxyNamespace) + 1));
+        // remove proxy namespace from class name
+        $className = substr($className, strlen($proxyNamespace));
```

Comment:
That comparison is a bit off, as the first var_dump includes the str_replace, while the second does not. We probably should compare this: https://3v4l.org/2vgog

Also, i'm not sure how relevant that is, as the slashes are removed with the str_replace in the next line anways.